djangocms_versioning/helpers.py

```diff
@@ -43,3 +44,28 @@ def replace_admin_for_models(models, admin_site=None):
         except KeyError:
             continue
         _replace_admin_for_model(modeladmin, admin_site)
+
+
+def register_versionadmin_proxy(version_proxy, grouper_name, admin_site=None):
+    """
+    ;param version_proxy: Proxy model to Version
+    :param grouper_name: Grouper model name
+    :param admin_site: AdminSite instance
+    """
+    if admin_site is None:
+        admin_site = admin.site
+
+    if version_proxy in admin_site._registry:
+        # Attempting to register the proxy again is a no-op.
+        return
+
+    class ProxiedAdmin(VersionAdmin):
+
+        def get_queryset(self, request):
+            content_type = ContentType.objects.get_for_model(self.model._content_model)
+            return super().get_queryset(request).filter(
+                content_type=content_type,
+            )
+    ProxiedAdmin.__name__ = grouper_name + VersionAdmin.__name__
```

Comment:
You're not really testing this line (the setting of the name) anywhere I think. I'd add an assert to one of your unit tests.